sycl/test/on-device/basic_tests/specialization_constants/non_native/gpu.cpp

```diff
@@ -1,4 +1,4 @@
-// REQUIRES: ocloc, gpu
+// REQUIRES: opencl, ocloc, gpu
```

Comment:
why did yo filter our Level Zero?

Reply:
>  AOT mode only, so Level Zero should not be here because it don't have such support.

Can you elaborate? What exactly do you think is not supported in Level Zero?

Reply:
Non-native spec constants are available currently only when user is compiling code in AOT mode: for OpenCL CPU/GPU/ACC and CUDA. Level Zero doesn't have ahead-of-time compilation support, only JITting, that's why this test for GPU device requires OpenCL only; for GPU device with CUDA backend this PR introduces another separate test.

Reply:
>  Level Zero doesn't have ahead-of-time compilation support

Do you mean SYCL RT running over Level Zero backend doesn't support AOT?
Is there anything missing in Level Zero itself to support AOT?

Reply:
OK, I had outdated information regarding Level Zero. Now I see that it really has a support of AOT when I compiled some example on Linux with Level Zero in AOT mode. I will remove `opencl` from the test and check CI again.

BTW, #3609 patch was merged, so currently checks for Linux and Windows are green in CI.